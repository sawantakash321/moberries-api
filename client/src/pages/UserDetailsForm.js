import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';
import React from 'react';

const UserDetailsForm = props => {
  const {
    values,
    handleChange,
    errors: {errors},
  } = props;

  return (
    <React.Fragment>
      <Typography variant="h6" gutterBottom>
        Enter User Details
      </Typography>
      <Grid container spacing={24}>
        <Grid item xs={12} sm={6}>
          <TextField
            required
            id="firstName"
            name="firstName"
            label="First name"
            fullWidth
            autoComplete="firstName"
            margin="normal"
            variant="outlined"
            onChange={handleChange('firstName')}
            defaultValue={values.firstName}
            error={values.firstName === ''}
            helperText={errors.errors.firstName}
          />
        </Grid>
        <Grid item xs={12} sm={6}>
          <TextField
            required
            id="lastName"
            name="lastName"
            label="Last name"
            fullWidth
            autoComplete="lastName"
            margin="normal"
            variant="outlined"
            onChange={handleChange('lastName')}
            defaultValue={values.lastName}
            error={values.lastName === ''}
            helperText={errors.errors.lastName}
          />
        </Grid>
        <Grid item xs={12}>
          <TextField
            required
            id="email"
            type="email"
            name="email"
            label="Email Address"
            fullWidth
            autoComplete="email"
            margin="normal"
            variant="outlined"
            onChange={handleChange('email')}
            defaultValue={values.email}
            error={values.email === ''}
            helperText={errors.errors.email}
          />
        </Grid>
        <Grid item xs={12}>
          <TextField
            required
            id="streetAddr"
            name="streetAddr"
            label="Street Address"
            fullWidth
            autoComplete="user-details address street-addr"
            margin="normal"
            variant="outlined"
            onChange={handleChange('streetAddr')}
            defaultValue={values.streetAddr}
            error={values.streetAddr === ''}
            helperText={errors.errors.streetAddr}
          />
        </Grid>
      </Grid>
    </React.Fragment>
  );
};

UserDetailsForm.propTypes = {
  values: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default UserDetailsForm;
