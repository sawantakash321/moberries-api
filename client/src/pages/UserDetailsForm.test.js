'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import UserDetailsForm from './UserDetailsForm';

// Snapshot test
it('matches the snapshot when there are no validation errors', () => {
  const values = {
    firstName: 'akash',
    lastName: 'sawant',
    email: 'sawantakash321@gmail.com',
    streetAddr: 'MLK street, behind cathom mart, Mumbai',
  };
  const errors = {
    errors: {
      firstName: '',
      lastName: '',
      email: '',
      streetAddr: '',
    },
  };
  const handleChange = jest.fn();
  const tree = renderer
    .create(<UserDetailsForm values={values} errors={{errors}} handleChange={handleChange} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
