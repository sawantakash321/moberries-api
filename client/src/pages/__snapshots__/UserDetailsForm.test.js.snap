// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot when there are no validation errors 1`] = `
Array [
  <h6
    className="MuiTypography-root-1 MuiTypography-h6-18 MuiTypography-gutterBottom-28"
  >
    Enter User Details
  </h6>,
  <div
    className="MuiGrid-container-37 MuiGrid-spacing-xs-24-61"
  >
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77 MuiGrid-grid-sm-6-85"
    >
      <div
        className="MuiFormControl-root-134 MuiFormControl-marginNormal-135 MuiFormControl-fullWidth-137"
      >
        <label
          className="MuiFormLabel-root-149 MuiFormLabel-required-154 MuiInputLabel-required-142 MuiInputLabel-root-138 MuiInputLabel-formControl-143 MuiInputLabel-animated-146 MuiInputLabel-outlined-148"
          data-shrink={false}
          htmlFor="firstName"
        >
          First name
          <span
            className="MuiFormLabel-asterisk-155"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-169 MuiOutlinedInput-root-156 MuiInputBase-fullWidth-178 MuiInputBase-formControl-170"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-186 MuiOutlinedInput-notchedOutline-163"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-187"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <input
            aria-invalid={false}
            autoComplete="firstName"
            className="MuiInputBase-input-179 MuiOutlinedInput-input-164"
            defaultValue="akash"
            disabled={false}
            id="firstName"
            name="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
        </div>
        
      </div>
    </div>
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77 MuiGrid-grid-sm-6-85"
    >
      <div
        className="MuiFormControl-root-134 MuiFormControl-marginNormal-135 MuiFormControl-fullWidth-137"
      >
        <label
          className="MuiFormLabel-root-149 MuiFormLabel-required-154 MuiInputLabel-required-142 MuiInputLabel-root-138 MuiInputLabel-formControl-143 MuiInputLabel-animated-146 MuiInputLabel-outlined-148"
          data-shrink={false}
          htmlFor="lastName"
        >
          Last name
          <span
            className="MuiFormLabel-asterisk-155"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-169 MuiOutlinedInput-root-156 MuiInputBase-fullWidth-178 MuiInputBase-formControl-170"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-186 MuiOutlinedInput-notchedOutline-163"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-187"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <input
            aria-invalid={false}
            autoComplete="lastName"
            className="MuiInputBase-input-179 MuiOutlinedInput-input-164"
            defaultValue="sawant"
            disabled={false}
            id="lastName"
            name="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
        </div>
        
      </div>
    </div>
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77"
    >
      <div
        className="MuiFormControl-root-134 MuiFormControl-marginNormal-135 MuiFormControl-fullWidth-137"
      >
        <label
          className="MuiFormLabel-root-149 MuiFormLabel-required-154 MuiInputLabel-required-142 MuiInputLabel-root-138 MuiInputLabel-formControl-143 MuiInputLabel-animated-146 MuiInputLabel-outlined-148"
          data-shrink={false}
          htmlFor="email"
        >
          Email Address
          <span
            className="MuiFormLabel-asterisk-155"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-169 MuiOutlinedInput-root-156 MuiInputBase-fullWidth-178 MuiInputBase-formControl-170"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-186 MuiOutlinedInput-notchedOutline-163"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-187"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <input
            aria-invalid={false}
            autoComplete="email"
            className="MuiInputBase-input-179 MuiOutlinedInput-input-164 MuiInputBase-inputType-182"
            defaultValue="sawantakash321@gmail.com"
            disabled={false}
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="email"
          />
        </div>
        
      </div>
    </div>
    <div
      className="MuiGrid-item-38 MuiGrid-grid-xs-12-77"
    >
      <div
        className="MuiFormControl-root-134 MuiFormControl-marginNormal-135 MuiFormControl-fullWidth-137"
      >
        <label
          className="MuiFormLabel-root-149 MuiFormLabel-required-154 MuiInputLabel-required-142 MuiInputLabel-root-138 MuiInputLabel-formControl-143 MuiInputLabel-animated-146 MuiInputLabel-outlined-148"
          data-shrink={false}
          htmlFor="streetAddr"
        >
          Street Address
          <span
            className="MuiFormLabel-asterisk-155"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-169 MuiOutlinedInput-root-156 MuiInputBase-fullWidth-178 MuiInputBase-formControl-170"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-186 MuiOutlinedInput-notchedOutline-163"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-187"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <input
            aria-invalid={false}
            autoComplete="user-details address street-addr"
            className="MuiInputBase-input-179 MuiOutlinedInput-input-164"
            defaultValue="MLK street, behind cathom mart, Mumbai"
            disabled={false}
            id="streetAddr"
            name="streetAddr"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
        </div>
        
      </div>
    </div>
  </div>,
]
`;
